assembly/src/tests.rs

```diff
@@ -203,7 +203,7 @@ fn script_with_import_errors() {
 #[test]
 fn comment_simple() {
     let assembler = super::Assembler::new();
-    let source = "begin # simple comment # push.1 push.2 add end";
+    let source = "begin push.1 push.2 add end # simple comment";
     let script = assembler.compile_script(source).unwrap();
```

Comment:
I think moving the comment to the end makes it a duplicate of the `comment_after_script` test below, so I would prefer if you can leave the comment inside the script and replace the second `#` with a line break instead (as you've done elsewhere)